ui/webui/test/README.rst

```diff
@@ -243,3 +266,60 @@ development environment that's independent from the host::
 
     toolbox create --image quay.io/cockpit/tasks -c cockpit
     toolbox enter cockpit
+
+Web UI integration tests
+========================
+
+Integration (or end to end) tests are stored in ``./ui/webui/test/integration``
+together with tplib test cases and test plans (those are only needed when running
+the tests using Permian).
+
+Integration tests use one more level of abstraction, to make it easier to write new
+tests and avoid code duplication class `IntegrationTest <https://github.com/rhinstaller/anaconda/blob/9baed94ac63fb2da73231ccc27ed88ac4360858b/ui/webui/test/helpers/integration.py#L38>`_
+should be used. This class handles the process of going through all the required
+installation steps with default options. So when writing new test you only have
+to use this class as parent and reimplement the functions that are important for
+the test case.
+
+Integration tests examples
+--------------------------
+
+There are three test cases in the anaconda repository that can be used as examples.
+
+**Default**
+
+Performs basically default installation.
+Test script ``default.py``, test case file ``default.tc.yaml``.
+
+**Storage encryption**
+
+Makes changes, compared to default installation, only in the storage section of
+the installation wizard and runs some commands before the system is rebooted.
+Test script ``storage_encryption.py``, test case file ``storage_encryption.tc.yaml``.
+
+**Wizard navigation**
+
+Changes the way how the test steps through the installation wizard.
+Test script ``wizard_navigation.py``, test case file ``wizard_navigation.tc.yam``.
+
+Running integration tests
+-------------------------
+
+Usually they are run by a Permian workflow and this is the recommended way,
+see documentation `here <https://permian.readthedocs.io/en/devel/workflows/anaconda-webui.html>`_.
+It is also possible to run them just by executing the test, but you will have to
+manually setup the environemnt to run them. See `Preparation and general invocation`_
+for unit tests.
+
+To run them completely, including the post-reboot checks, they have to be run on
+an existing machine that is running the installer with cmdline options ``inst.sshd inst.webui.remote``
+(VM spawned by cockpit framework won't survive reboot).
```

Comment:
I think this is redundant. cockpit VMs are transient and they don't survive shutdown indeed, but someone should not use cockpit commands to spawn Vms not used for cockpit library based testing anyway.

Reply:
I'm not sure what do you mean. Even the end to end tests use cockpit testing library and if they are executed on their own they will try to spawn VM. Someone trying to run anaconda test may not be familiar with the cockpit testing framework, thats why I wrote it here.